tests/accumulate.rs

```diff
@@ -0,0 +1,173 @@
+// copyright 2016 maidsafe.net limited.
+//
+// This SAFE Network Software is licensed to you under (1) the MaidSafe.net Commercial License,
+// version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
+// licence you accepted on initial access to the Software (the "Licences").
+//
+// By contributing code to the SAFE Network Software, or to this project generally, you agree to be
+// bound by the terms of the MaidSafe Contributor Agreement, version 1.1.  This, along with the
+// Licenses can be found in the root directory of this project at LICENSE, COPYING and CONTRIBUTOR.
+//
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.
+//
+// Please review the Licences for the specific language governing permissions and limitations
+// relating to use of the SAFE Network Software.
+
+extern crate itertools;
+#[macro_use]
+extern crate log;
+extern crate rand;
+extern crate routing;
+#[macro_use]
+extern crate unwrap;
+
+// use itertools::Itertools;
+// use rand::Rng;
+// use routing::{Authority, Cache, NullCache, Client};
+// use routing::{Data, DataIdentifier, ImmutableData};
+// use routing::{Event, FullId};
+// use routing::{Request, Response};
+// use routing::mock_crust::{self, Config, Endpoint, Network, ServiceHandle};
+// use routing::mock_crust::crust::PeerId;
+// use routing::Node;
+// use routing::MIN_GROUP_SIZE;
+// use routing::{Prefix, Xorable};
+// use routing::{Destination, RoutingTable, verify_network_invariant};
+// use routing::MessageId;
+// use routing::XorName;
+// use routing::mock_crust::utils::*;
+// use std::cell::RefCell;
+// use std::cmp;
+// use std::collections::{HashMap, HashSet};
+// use std::sync::mpsc;
+// 
+// /// Expect that the next event raised by the node matches the given pattern.
+// /// Panics if no event, or an event that does not match the pattern is raised.
+// /// (ignores ticks).
+// macro_rules! expect_next_event {
```

Comment:
Rather than duplicating `expect_next_event!`, `expect_any_event!` and `expect_no_event!`, I'd prefer they were exported from /src/mock_crust/utils.rs.